radix-engine-lib/src/engine/api.rs

```diff
@@ -0,0 +1,111 @@
+use crate::crypto::Hash;
+use crate::data::ScryptoCustomTypeId;
+use crate::engine::actor::ScryptoActor;
+use crate::engine::scrypto_env::NativeFnInvocation;
+use crate::model::*;
+use sbor::rust::fmt::Debug;
+use sbor::rust::string::String;
+use sbor::rust::vec::Vec;
+use sbor::Decode;
+
+use super::types::*;
+
+pub trait ScryptoNativeInvocation: Into<NativeFnInvocation> + SysInvocation {}
+
+pub trait SysInvocation {
```

Comment:
We can remove the `Sys` prefix, so we have:
- `Invocation`
- `ScryptoNativeInvocation`
- `NativeInvokable`
- `NativeInvokables` - the set of invkables